### PR TITLE
create docs for synthetic data generator gem

### PR DESCRIPTION
**Pull request by atbida**
Todo: 
1. incorporate Alex's comments (DONE), and mention the file generated is saved as any file type option in the target type&format modal. (DONE)
2. update the images to reflect the new sample synthetic data <--- generate new synthetic data with UIDs per Maciej DONE
3. include the requirements for spark version etc. from the coda! DONE
4. consider using the staging environment once Sashank has updated some of [these](https://app.asana.com/0/1203813959485797/1208287989763800/f) items, and see if I can get the databricks cluster to stay attached in staging.  DONE

### CODE REVIEW

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/synthetic-data-generator.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Create the target Gem.
+![img](../../../img/synth_4_new_target.png)
+
+Connect the Synthetic Data Generator SOURCE Gem to the Target Gem.
+![img](../../../img/synth_5_connect_target.png)
+
+Be sure to configure the write mode for the target Gem. This is **very important** because the Synthetic Data Generator Gem is **not** idempotent. There is a **new seed** each time the Gem is run.
+![img](../../../img/synth_6_write_mode.png)
+
+:::caution
```
Comment (atbida) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/synthetic-data-generator.md:52]: related question[ here](https://app.asana.com/0/0/1207327805824002/1208262293543780/f)

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/synthetic-data-generator.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  - random
+  - fake
+  - data
+  - generator
+---
+
+Generate synthetic data with this special kind of Source Gem.
+
+Generating mock data is crucial when building data Pipelines to simulate real-world scenarios for testing, validating, and optimizing Pipeline performance before using actual production data. It helps ensure the Pipeline handles various data formats, structures, and edge cases effectively, minimizing potential issues in a live environment.
+
```
Comment (atbida) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/synthetic-data-generator.md:17]: Let's add a fun dataset example to show the "why."

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/synthetic-data-generator.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+![img](../../../img/synth_4_new_target.png)
+
+Connect the Synthetic Data Generator SOURCE Gem to the Target Gem.
+![img](../../../img/synth_5_connect_target.png)
+
+Be sure to configure the write mode for the target Gem. This is **very important** because the Synthetic Data Generator Gem is **not** idempotent. There is a **new seed** each time the Gem is run.
+![img](../../../img/synth_6_write_mode.png)
+
+:::caution
+The Synthetic Data Generator only generates the data. If you want to store the data just connect the output to a target Gem and configure the location, write properties etc. In particular, consider the data is generated again for each run. The target write mode can be error, overwrite, append, or ignore as desired.
```
Comment (alexanderahn) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/synthetic-data-generator.md:59]: "In particular, consider the data is generated again for each run." I'm not sure what the phrasing of this means. Do we simply want to say, "The data is generated again for each run."?

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  - generator
+---
+
+To generate a new column of random data, select one of the [providers](#providers) below. There are some properties [common](#common-properties) to all providers. If you prefer, provide the same information as a JSON schema.
+
+## Providers
+
+| Data Provider             | Description                                                                                                                                                                                                                                                 |
+| ------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Random Name               | Generates random names. Select Full Name, First Name, Last Name as the sub-types.                                                                                                                                                                           |
```
Comment (alexanderahn) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md:22]: Should we add "and" here?

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+| Random Address            | Generates random addresses                                                                                                                                                                                                                                  |
+| Random Email              | Generates random emails                                                                                                                                                                                                                                     |
+| Random Phone Number       | Generates random phone numbers based on specified or default pattern. Example: specify the pattern for a phone number as (###) ###-####                                                                                                                     |
+| Random String UUID        | Generates random uuids in string form.                                                                                                                                                                                                                      |
```
Comment (alexanderahn) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md:26]: Should uuids be UUIDs?

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+| Random Boolean Values     | Generates random boolean values (True/False). TODO                                                                                                                                                                                                          |
+| Random Integer Numbers    | Generates random integers within the range from Start Value to End Value. TODO                                                                                                                                                                              |
+| Random Elements From List | Generates random values from the list of values. Just type into the `List Of Values` feild. TODO                                                                                                                                                            |
```
Comment (alexanderahn) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md:29]: typo: "field"

File: docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## Providers
+
+| Data Provider             | Description                                                                                                                                                                                                                                                 |
+| ------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Random Name               | Generates random names. Select Full Name, First Name, Last Name as the sub-types.                                                                                                                                                                           |
+| Random Address            | Generates random addresses                                                                                                                                                                                                                                  |
```
Comment (alexanderahn) [inline docs/Spark/gems/source-target/advanced/synthetic-data-generator/providers.md:23]: Some of the ends of these lines have periods. Others do not. I suggest adding periods for all.

**Review by alexanderahn — APPROVED** [review]
Thank you, Anya, for this. I like the structure of the pages you created. I left a few comments where I think some rewrites would help. Otherwise, LGTM!

**Pull request merged**